reddit.js

```diff
@@ -0,0 +1,204 @@
+{
+	"translatorID": "23bacc11-98e3-4b78-b1ef-cc2c9a04b893",
+	"label": "reddit",
+	"creator": "Lukas Kawerau",
+	"target": "^https?://www\\.reddit\\.com",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-10-29 08:53:37"
+}
+
+/*
+***** BEGIN LICENSE BLOCK *****
+Reddit Translator
+Copyright (C) 2020 Lukas Kawerau, lukas@kawerau.org
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+   ***** END LICENSE BLOCK *****
+*/
+
+
+// attr()/text() v2
+// eslint-disable-next-line
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null}
+
+function detectWeb(doc, url) {
+	if (url.includes('search') && getSearchResults(doc, true)) {
+		return 'multiple';
+	} 
+	else if (url.match(/\/r\/[a-z\d]+\/comments\//g)) {
+		return 'forumPost';
+	}  else {
+		return 'multiple';
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.SQnoC3ObvgnGjWt90zD9Z._2INHSNB8V5eaWp4P0rY_mE');
```

Comment:
Yes - they seem to be stable across searches and browsers. I've tested across Chrome and Safari, including private windows and different searches on different subreddits - always stable.

Reply:
They still might be generated at build time, meaning a site update could break them. I haven't looked at the page to see what else is available, but something less random would certainly be preferable.

Reply:
Hm, I see. I'm not skilled enough to find a less random identifier, so if this is blocking the PR I'd appreciate any help in finding a more suitable way of going about this.

Reply:
I think this would only work with xpaths since you need a parent selector. `h3` is the only clearly distinguishing feature (no useful attributes anywhere) and the link is in `a` two levels up. I'd have to test but something like
`var rows = ZU.xpath(doc, '//a[div/h3]`); looks like it should work

Reply:
Seems to work!